bundles/org.openhab.binding.guntamatic/src/main/java/org/openhab/binding/guntamatic/internal/GuntamaticHandler.java

```diff
@@ -0,0 +1,426 @@
+                    case CHANNEL_CONTROLPROGRAM:
+                        param = getThing().getProperties().get(PARAMETER_PROGRAM);
+                        break;
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM0:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM1:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM2:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM3:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM4:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM5:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM6:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM7:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM8:
+                        param = getThing().getProperties().get(PARAMETER_HEATCIRCPROGRAM).replace("x",
+                                channelID.substring(channelID.length() - 1));
+                        break;
+                    case CHANNEL_CONTROLWWHEAT0:
+                    case CHANNEL_CONTROLWWHEAT1:
+                    case CHANNEL_CONTROLWWHEAT2:
+                        param = getThing().getProperties().get(PARAMETER_WWHEAT).replace("x",
+                                channelID.substring(channelID.length() - 1));
+                        break;
+                    case CHANNEL_CONTROLEXTRAWWHEAT0:
+                    case CHANNEL_CONTROLEXTRAWWHEAT1:
+                    case CHANNEL_CONTROLEXTRAWWHEAT2:
+                        param = getThing().getProperties().get(PARAMETER_EXTRAWWHEAT).replace("x",
+                                channelID.substring(channelID.length() - 1));
+                        break;
+                    default:
+                        return;
+                }
+                String response = sendGetRequest(PARSET_URL, "syn=" + param + "&value=" + command.toString());
+                // logger.warn("{} syn={}&value={}", PARSET_URL, param, command.toString());
+                State newState = new StringType(response);
+                updateState(channelID, newState);
+            } else {
+                logger.warn("A 'key' needs to be configured in order to control the Guntamatic Heating System");
+            }
+        }
+    }
+
+    private void parseAndUpdate(String html) {
+        String[] daqdata = html.split("\\n");
+
+        for (Integer i : channels.keySet()) {
+            String channel = channels.get(i);
+            String unit = units.get(i);
+            if ((channel != null) && (unit != null)) {
+                String value = daqdata[i].trim();
+                Channel chn = thing.getChannel(channel);
+                if ((chn != null) && ("Switch".equals(chn.getAcceptedItemType()))) {
+                    // Guntamatic uses German OnOff when configured to German and English OnOff for all other languages
+                    value = value.replace("AUS", "OFF").replace("EIN", "ON");
+                }
+                State newState = new StringType(value + unit);
```

Comment:
You should create the proper state, OnOffType, QuantityType, ...

Reply:
What's the benefit?
According to my understanding, that would require a quite inflexible code monster. The implicite cast works like a charme.

Reply:
May I use `TypeParser.parseState`?
[TypeParser.parseState](https://www.openhab.org/javadoc/latest/org/openhab/core/types/typeparser)

Reply:
The following implementation is compiling, but not yet tested:

```
                    String typeName = chn.getAcceptedItemType();
                    if ("Switch".equals(typeName)) {
                        // Guntamatic uses German OnOff when configured to German and English OnOff for all other
                        // languages
                        if ("ON".equals(value) || "EIN".equals(value)) {
                            updateState(channel, OnOffType.ON);
                        } else if ("OFF".equals(value) || "AUS".equals(value)) {
                            updateState(channel, OnOffType.OFF);
                        }
                    } else {
                        State newState = null;
                        if ("Number".equals(typeName)) {
                            newState = new DecimalType(value);
                        } else if ("Number:Dimensionless".equals(typeName)) {
                            if ("%".equals(unit)) {
                                newState = new QuantityType<Dimensionless>(Double.parseDouble(value), Units.PERCENT);
                            }
                        } else if ("Number:Temperature".equals(typeName)) {
                            if ("°C".equals(unit)) {
                                newState = new QuantityType<Temperature>(Double.parseDouble(value), SIUnits.CELSIUS);
                            } else if ("°F".equals(unit)) {
                                newState = new QuantityType<Temperature>(Double.parseDouble(value),
                                        ImperialUnits.FAHRENHEIT);
                            }
                        } else if ("Number:Volume".equals(typeName)) {
                            if ("m³".equals(unit)) {
                                newState = new QuantityType<Volume>(Double.parseDouble(value) * 1000.0, Units.LITRE);
                            }
                        } else if ("Number:Time".equals(typeName)) {
                            if ("d".equals(unit)) {
                                newState = new QuantityType<Time>(Double.parseDouble(value), Units.DAY);
                            } else if ("h".equals(unit)) {
                                newState = new QuantityType<Time>(Double.parseDouble(value), Units.HOUR);
                            }
                        } else if ("String".equals(typeName)) {
                            newState = new StringType(value);
                        }

                        if (newState != null) {
                            updateState(channel, newState);
                        }
                    }
```

I'm still not sure, if this is an improvement to the original implementation.

Reply:
> ..., but not yet tested

In the meanwhile I added exception handling for `parseDouble` and tested the new implementation successfully.

Reply:
I will propose to you a way to simplify this.